protostar/starknet/cheaters/stateful.py

```diff
@@ -0,0 +1,69 @@
+import copy
+import collections
+
+from abc import ABC
+from typing import Optional, TYPE_CHECKING
+from typing_extensions import Self
+
+from starkware.python.utils import to_bytes, from_bytes
+
+from protostar.starknet.cheater import Cheater, CheaterException
+from protostar.starknet.address import Address
+from protostar.starknet.data_transformer import (
+    DataTransformerException,
+    CairoOrPythonData,
+    CairoData,
+    from_python_transformer,
+)
+
+if TYPE_CHECKING:
+    from protostar.starknet.cheatable_cached_state import CheatableCachedState
+
+
+class StatefulCheater(Cheater, ABC):
+    def __init__(self, cheatable_state: "CheatableCachedState"):
+        self.cheatable_state = cheatable_state
+
+    def copy(self) -> Self:
+        return copy.copy(self)
+
+    def apply(self, parent: Self) -> None:
+        pass
+
+    async def _transform_calldata_to_cairo_data_by_addr(
```

Comment:
TBH I think this class is a bit messy, 
`_transform_calldata_to_cairo_data_by_addr`
`_transform_calldata_to_cairo_data`
are just a bit random logic dump touching different parts of application.
I still think we should add
`get_contract_class_by_address` to cheatable state or separate function. If we need some transformation logic abstraction we should add it to the transformer.

Reply:
Do you feel strongly about this? Or can we address this separately?

I don't find this messy, this function ("overloaded") is pretty generic and should be in such a mutual place - it is used by both, contract and expects cheaters.

Reply:
I also don't fully understand what you said :/ Are there 2 separate parts of your comments or are they related? How adding `get_contract_class_by_address` is connected to the fact that `_transform_calldata_to_cairo_data` is where it is now?

Reply:
As I said, I'm not sure if I understood well what you meant. Could you create an issue? I'd do it but I'm afraid I'm not going to attach all the information in the proper way.

Reply:
Ok, explained offline, [move transform_calldata_to_cairo_data](https://github.com/software-mansion/protostar/pull/1363/commits/1ce5ae1de8a3f05f51ed99b1e898aed3d9bd50fd)